src/main/kotlin/uk/gov/dluhc/printapi/database/entity/Certificate.kt

```diff
@@ -95,30 +100,98 @@ class Certificate(
         return this
     }
 
-    fun getCurrentPrintRequest(): PrintRequest {
-        printRequests.sortByDescending { it.requestDateTime }
-        return printRequests.first()
+    fun getPrintRequestsByStatus(printRequestStatus: Status) =
+        printRequests.filter { it.getCurrentStatus().status == printRequestStatus }
+
+    fun addPrintRequestToBatch(batchId: String) {
+        getPrintRequestsByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH).forEach {
+            it.addPrintRequestStatus(
+                PrintRequestStatus(
+                    status = Status.ASSIGNED_TO_BATCH,
+                    eventDateTime = Instant.now(),
+                    message = null
+                )
+            )
+            it.batchId = batchId
+        }
+        assignStatus()
     }
 
-    /**
-     * Adds the new status to the current PrintRequest and updates the current Certificate status.
-     */
-    fun addStatus(
+    fun addSentToPrintProviderEventForBatch(batchId: String) {
+        getPrintRequestsByBatchId(batchId).forEach {
+            it.addPrintRequestStatus(
+                PrintRequestStatus(
+                    status = Status.SENT_TO_PRINT_PROVIDER,
+                    eventDateTime = Instant.now(),
+                    message = null
+                )
+            )
+            it.batchId = batchId
```

Comment:
@Neil-Massey , is this needed here, since we iterate through the printRequests with the batchId equal to this `batchId`?

Reply:
Good spot - will remove 👍